feat: [IDP-3558]: Added docs for the newly released ingestion api (#7916)
harness/developer-hub: docs/internal-developer-portal/catalog/custom-catalog-properties.md

## API Examples and Usage

> ## Entities to change/update are Specified or Known

Reviewer: ```suggestion
## Use-case: When the entities to update are specified
```

> ## Catalog Metadata Ingestion API for given entity for a single property

Reviewer: too many H2s here. Looks odd when rendered.

- **mode:**`mode` is optional, takes value `append/replace` and is in use when the property holds an **array** like `tags`, `relations`, etc. For arrays, the **default** mode value is `replace` e.g. if you post a new tag, it will replace the existing tags added in the entity.

- **value:** This field contains the value of the entity added under `field`.

![](./static/inspect-entity.png)

![](./static/raw-yaml.png)

```YAML
###Example Processed Entitiy YAML as displayed in IDP

apiVersion: backstage.io/v1alpha1
kind: Component
metadata:
  name: my-new-service
  description: Description of my new service
  annotations:
       harness.io/project-url: https://app.harness.io/ng/account/vpCkHK/module/idp-admin/orgs/default/projects/Backstage/pipelines/Releasenpm/
  cloudCost: 24$
  codeCoverageScore: 83
...
```

> ## Catalog Metadata Ingestion API to ingest single property across selected properties

Reviewer: ```suggestion
## Catalog Metadata Ingestion API to ingest single property across selected entities
```
